package/base-files/files/lib/functions/ipv4.sh

```diff
@@ -0,0 +1,142 @@
+#
+# rewrite of /bin/ipcalc.sh from awk to shell for use as a library
+# in init.d wrappers, and better portability (since busybox awk
+# lacks -E support at this time).
+#
+
+decimal=0
+
+bitcount() {
+    local c="$1"
+
+    c=$((((c >> 1) & 0x55555555) + (c & 0x55555555)))
+    c=$((((c >> 2) & 0x33333333) + (c & 0x33333333)))
+    c=$((((c >> 4) & 0x0f0f0f0f) + (c & 0x0f0f0f0f)))
+    c=$((((c >> 8) & 0x00ff00ff) + (c & 0x00ff00ff)))
+    c=$((((c >> 16) & 0x0000ffff) + (c & 0x0000ffff)))
+
+    echo "$c"
+}
+
+int32() {
```

Comment:
This needs another name.
Since it's doing `exit 1`, it's basically an assertion.
How about `assert_int32`?

Reply:
also, `uint32` - no signs allowed.

Reply:
Changed to `check_uint32()`.